File: 1829 Maximum XOR for Each Query/1829.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int maxValue = pow(2, maximumBit) - 1;
        int cur = 0;

        for (int num: nums){
            cur ^= num;
        }

        vector<int> ans;
        int n = nums.size();
        
        for (int i = n-1; i>=0; i--){
            int k = maxValue ^ cur;
            ans.push_back(k);
            cur ^= nums[i];
        }
        
        return ans;
    }
};

File: 1829 Maximum XOR for Each Query/1829.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        max_value = 2**maximumBit - 1
        cur = 0

        for num in nums:
            cur ^= num

        ans = []
        n = len(nums)
        for num in nums[::-1]:
            k = max_value ^ cur
            ans.append(k)
            cur ^= num

        return ans
